### PR TITLE
[Merged by Bors] - Feat/template tests

### PR DESCRIPTION
**Pull request by razvan**
## Description

Added test templates. Works together with: https://github.com/stackabletech/operator-templating/pull/110

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: tests/templates/kuttl/spark-pi-public-s3/test.j2 [hunk 1]
```diff
@@ -0,0 +1 @@
+{{ test_scenario }}
```
Comment (soenkeliebau) [inline tests/templates/kuttl/spark-pi-public-s3/test.j2:1]: I think this is leftover from development and can be removed.

File: tests/test-definition.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - name: spark-pi-private-s3
+    dimensions:
+      - spark
+      - hadoop
+      - stackable
+  - name: spark-pi-public-s3
+    dimensions:
+      - spark
+      - hadoop
+      - stackable
```
Comment (soenkeliebau) [inline tests/test-definition.yaml:21]: Missing newline

File: tests/test-definition.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+dimensions:
```
Comment (soenkeliebau) [inline tests/test-definition.yaml:1]: Missing ---

**Review by soenkeliebau — CHANGES_REQUESTED** [review]
minor stuff

Disclaimer: didn't review the test cases themselves, I assume those have been reviewed in the originating repo

File: tests/templates/kuttl/spark-pi-public-s3/test.j2 (hunk 1 above)
Comment (razvan) [reply tests/templates/kuttl/spark-pi-public-s3/test.j2]: Removed,

File: tests/test-definition.yaml (hunk 2 above)
Comment (razvan) [reply tests/test-definition.yaml]: Formatted.

**Review by soenkeliebau — APPROVED** [review]

**Pull request closed without merging**